### PR TITLE
Add unit test to expose undeterministic merge with empty string key

### PR DESCRIPTION
**Pull request by csrwng**
Exposes undeterministic merge introduced in mergo v0.3.9

To test locally...
`go get github.com/imdario/mergo@v0.3.9` and run this test

### CODE REVIEW

File: pkg/operator/configobserver/config_observer_controller_test.go [hunk 1]
```diff
@@ -112,7 +115,7 @@ func TestSyncStatus(t *testing.T) {
 
 			expectError: false,
 			expectedObservedConfig: &unstructured.Unstructured{Object: map[string]interface{}{
-				"foo": "one",
+				"foo": map[string]interface{}{"one": "1", "two": ""},
```
Comment (deads2k) [inline pkg/operator/configobserver/config_observer_controller_test.go:118]: so what happens on the latest mergo?

Comment (csrwng) [reply pkg/operator/configobserver/config_observer_controller_test.go:118]: on 0.3.11 it passes

Comment (csrwng) [reply pkg/operator/configobserver/config_observer_controller_test.go:118]: so presumably it's ok :) ?
